src/cargo-fmt/main.rs

```diff
@@ -322,15 +322,18 @@ fn run_rustfmt(
     fmt_args: &[String],
     verbosity: Verbosity,
 ) -> Result<i32, io::Error> {
-    let by_edition: HashMap<_, _> = targets
+    let by_edition = targets
```

Comment:
The fix looks solid to me, but I couldn't help thinking writing this snippet in imperative style would make it a bit shorter:
```
    let mut by_edition = HashMap::new();

    for t in targets {
        if verbosity == Verbosity::Verbose {
            println!("[{} ({})] {:?}", t.kind, t.edition, t.path)
        }
        by_edition.entry(&t.edition).or_insert_with(Vec::new).push(&t.path);
    }
```
Of course, this is purely a matter of personal preference, the code is perfectly fine as is.

Reply:
Not  really, the code formatted looks like:

```rust
    let mut by_edition = HashMap::new();
    for t in targets {
        if verbosity == Verbosity::Verbose {
            println!("[{} ({})] {:?}", t.kind, t.edition, t.path)
        }
        by_edition
            .entry(&t.edition)
            .or_insert_with(Vec::new)
            .push(&t.path);
    }
```
vs
```rust
    let by_edition: HashMap<_, _> = targets
        .iter()
        .inspect(|t| {
            if verbosity == Verbosity::Verbose {
                println!("[{} ({})] {:?}", t.kind, t.edition, t.path)
            }
        })
        .map(|t| (&t.edition, vec![&t.path]))
        .collect();
```

Reply:
Fair enough, it's always better to avoid `mut`.